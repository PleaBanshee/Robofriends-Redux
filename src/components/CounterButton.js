import React, {useState} from 'react';

function CounterButton({color}) {
    const [count,setCount] = useState(0)

    const updateCount = () => {
        setCount(count+1);
    }

    console.log({color,count});
    return <button color={color} onClick={updateCount}>Count: {count}</button>
}

// NB! React.memo keeps a component from receiving state, so don't use it on components that needs state updates
export default (CounterButton);