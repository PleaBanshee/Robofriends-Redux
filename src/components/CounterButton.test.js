import { shallow } from 'enzyme';
import CounterButton from './CounterButton';
  
it('The counter button renders', () => {
    expect.assertions(1);
    const mockColor = 'red';
    expect(shallow(<CounterButton color={mockColor}/>)).toMatchSnapshot();
});

it('The counter increments by one', () => {
    expect.assertions(3);
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor}/>);
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.find('[id="counter"]').text()).toEqual('Count: 2');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.find('[id="counter"]').text()).toEqual('Count: 3');
    expect(wrapper.props().color).toEqual('red');
});