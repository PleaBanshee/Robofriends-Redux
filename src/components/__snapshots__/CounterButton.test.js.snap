// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The counter button renders 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
